File: docs/breez-sdk/snippets/python/src/lnurl_auth.py
import logging
from breez_sdk_spark import (
    BreezSdk,
    InputType,
    LnurlAuthRequestDetails,
    LnurlCallbackStatus,
)


async def parse_lnurl_auth(sdk: BreezSdk):
    # ANCHOR: parse-lnurl-auth
    # LNURL-auth URL from a service
    # Can be in the form:
    # - lnurl1... (bech32 encoded)
    # - https://service.com/lnurl-auth?tag=login&k1=...
    lnurl_auth_url = "lnurl1..."

    try:
        input_type = await sdk.parse(lnurl_auth_url)
        if isinstance(input_type, InputType.LNURL_AUTH):
            request_data = input_type[0]
            logging.debug(f"Domain: {request_data.domain}")
            logging.debug(f"Action: {request_data.action}")

            # Show domain to user and ask for confirmation
            # This is important for security
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: parse-lnurl-auth


async def authenticate(sdk: BreezSdk, request_data: LnurlAuthRequestDetails):
    # ANCHOR: lnurl-auth
    # Perform LNURL authentication
    try:
        result = await sdk.lnurl_auth(request_data=request_data)

        if isinstance(result, LnurlCallbackStatus.OK):
            logging.debug("Authentication successful")
        elif isinstance(result, LnurlCallbackStatus.ERROR_STATUS):
            logging.debug(f"Authentication failed: {result.error_details.reason}")
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: lnurl-auth

File: docs/breez-sdk/snippets/rust/src/lnurl_auth.rs
use anyhow::Result;
use breez_sdk_spark::*;
use log::info;

async fn parse_lnurl_auth(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: parse-lnurl-auth
    // LNURL-auth URL from a service
    // Can be in the form:
    // - lnurl1... (bech32 encoded)
    // - https://service.com/lnurl-auth?tag=login&k1=...
    let lnurl_auth_url = "lnurl1...";

    if let Ok(InputType::LnurlAuth(request_data)) = sdk.parse(lnurl_auth_url).await {
        info!("Domain: {}", request_data.domain);
        info!("Action: {:?}", request_data.action);

        // Show domain to user and ask for confirmation
        // This is important for security
    }
    // ANCHOR_END: parse-lnurl-auth
    Ok(())
}

async fn authenticate(sdk: &BreezSdk, request_data: LnurlAuthRequestDetails) -> Result<()> {
    // ANCHOR: lnurl-auth
    // Perform LNURL authentication
    let result = sdk.lnurl_auth(request_data).await?;

    match result {
        LnurlCallbackStatus::Ok => {
            info!("Authentication successful");
        }
        LnurlCallbackStatus::ErrorStatus { error_details } => {
            info!("Authentication failed: {}", error_details.reason);
        }
    }
    // ANCHOR_END: lnurl-auth
    Ok(())
}
